barretenberg/cpp/src/barretenberg/benchmark/ultra_bench/mock_proofs.hpp

```diff
@@ -47,46 +47,27 @@ template <typename Builder> void generate_basic_arithmetic_circuit(Builder& buil
     }
 }
 
-// ultrahonk
-inline UltraProver get_prover(UltraComposer& composer,
-                              void (*test_circuit_function)(UltraComposer::CircuitBuilder&, size_t),
-                              size_t num_iterations)
+template <typename Prover>
+Prover get_prover(void (*test_circuit_function)(typename Prover::Flavor::CircuitBuilder&, size_t),
+                  size_t num_iterations)
 {
-    UltraComposer::CircuitBuilder builder;
-    test_circuit_function(builder, num_iterations);
-    std::shared_ptr<UltraComposer::ProverInstance> instance = composer.create_prover_instance(builder);
-    return composer.create_prover(instance);
-}
-
-inline GoblinUltraProver get_prover(GoblinUltraComposer& composer,
-                                    void (*test_circuit_function)(GoblinUltraComposer::CircuitBuilder&, size_t),
-                                    size_t num_iterations)
-{
-    GoblinUltraComposer::CircuitBuilder builder;
-    test_circuit_function(builder, num_iterations);
-    std::shared_ptr<GoblinUltraComposer::ProverInstance> instance = composer.create_prover_instance(builder);
-    return composer.create_prover(instance);
-}
+    using Flavor = typename Prover::Flavor;
+    using Builder = typename Flavor::CircuitBuilder;
 
-// standard plonk
-inline plonk::Prover get_prover(plonk::StandardComposer& composer,
-                                void (*test_circuit_function)(StandardCircuitBuilder&, size_t),
-                                size_t num_iterations)
-{
-    StandardCircuitBuilder builder;
+    Builder builder;
     test_circuit_function(builder, num_iterations);
-    return composer.create_prover(builder);
-}
+    // This is gross but it's going away soon.
```

Comment:
soon, from what? Plonk getting removed?

Reply:
Yeah

Reply:
Once we have a UH smart contract. Sean is working on this.